pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -404,12 +404,12 @@ func (c *Controller) deleteService(svc *model.Service) {
 	c.Lock()
 	delete(c.servicesMap, svc.Hostname)
 	delete(c.nodeSelectorsForServices, svc.Hostname)
+	c.Unlock()
+
 	c.networkManager.Lock()
 	_, isNetworkGateway := c.networkGatewaysBySvc[svc.Hostname]
 	delete(c.networkGatewaysBySvc, svc.Hostname)
```

Comment:
Wouldn't this call needs c.Lock?

Reply:
Not need , `networkManager` is not recreated. Any update to it can be treated as a separate op